### PR TITLE
[3546] Channel list loads stale data from the DB

### PR DESCRIPTION
**Pull request by kanat**
Closes: https://github.com/GetStream/stream-chat-android/issues/3546
Related PR: https://github.com/GetStream/stream-chat-android/pull/3524

### 🎯 Goal

Show the latest actual data in offline mode.

### 🧪 Testing

1. Open either sample app while online
2. Stay on the channel list
3. Take a screenshot of the channel list
4. Kill the app
5. Go offline
6. Re-open the app while offline and stay on the channel list
7. Take a screenshot
8. Compare screenshots

Expected: The channel list shows the latest previously seen online state.

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch
- [x] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [x] Changelog is updated with client-facing changes
- [x] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### 🎉 GIF

_Please provide a suitable gif that describes your work on this pull request_

### CODE REVIEW

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/event/handler/internal/EventHandlerSequential.kt [hunk 1]
```diff
@@ -449,8 +448,13 @@ internal class EventHandlerSequential(
...
                     batch.addMessageData(event.cid, event.message, isNewMessage = true)
-                    repos.selectChannelWithoutMessages(event.cid)?.copy(hidden = false)
-                        ?.let(batch::addChannel)
+                    repos.selectChannelWithoutMessages(event.cid)?.let { channel ->
+                        val updatedChannel = channel.copy(
+                            hidden = false,
+                            messages = listOf(event.message)
+                        )
+                        batch.addChannel(updatedChannel)
+                    }
```
Comment (kanat) [inline stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/event/handler/internal/EventHandlerSequential.kt:457]: The same actual **N1** fix for the sequential implementation.

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/event/handler/internal/batch/BatchEvent.kt [hunk 2]
```diff
@@ -17,11 +17,14 @@
...
 
 import io.getstream.chat.android.client.events.ChatEvent
+import kotlin.math.absoluteValue
+import kotlin.random.Random
 
 /**
  * Events container to represent the source of the received events.
  */
 internal class BatchEvent(
+    val id: Int = Random.nextInt().absoluteValue,
```
Comment (MarinTolic) [inline stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/event/handler/internal/batch/BatchEvent.kt:27]: Do these strictly have to be unique or not?
I'm wondering if there's a chance of collision.

**Review by MarinTolic — APPROVED** [review]
Works beautifully. :)
Please include a gif next time, it makes us happy. :(

### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [x] Compose sample runs & works
- [x] Bugs validated (bugfixes)
- [x] New feature tested and works
- [x] Release notes and docs clearly describe changes
- [x] All code we touched has new or updated KDocs

**Review by leandroBorgesFerreira — APPROVED** [review]

**Comment by leandroBorgesFerreira** [comment]
### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [x] Compose sample runs & works
- [x] Bugs validated (bugfixes)
- [x] New feature tested and works
- [x] Release notes and docs clearly describe changes
- [x] All code we touched has new or updated KDocs

**Pull request merged**